src/uucore/src/lib/features/checksum.rs

```diff
@@ -75,6 +75,65 @@ struct ChecksumResult {
     pub failed_open_file: i32,
 }
 
+/// Represents a reason for which the processing of a checksum line
+/// could not proceed to digest comparison.
+enum LineCheckError {
+    /// a generic UError was encountered in sub-functions
+    UError(Box<dyn UError>),
+    /// the line is empty or is a comment
+    Skipped,
+    /// the line has a formatting error
+    ImproperlyFormatted,
+    /// the given file is not found, may be a directory
+    CantOpenFile(bool),
```

Comment:
What's the meaning of the `bool` field?

Reply:
Here, the boolean represents the fact that the error should increment the `failed_open_file` of the `ChecksumResult` structure.
Missing files will increment this file if the flag `--ignore-missing` was not provided (otherwise this is considered as a silent error), but a file actually being a directory will trigger an increment in any case

Reply:
Hm, I think you are mixing concerns because the field is not about the error variant itself but about what the error handler should do if it encounters this error variant.

As the `CantOpenFile` variant is returned for three different reasons (`File::open()` fails, `metadata()` fails, file is a directory), it might make sense to have a variant for each reason.

Reply:
Sure, I can do this :+1: